prealloc.go

```diff
@@ -215,13 +226,33 @@ func (v *returnsVisitor) Visit(node ast.Node) ast.Visitor {
 					if !ok {
 						continue
 					}
-					if genD.Tok == token.VAR {
+					if genD.Tok == token.TYPE {
```

Comment:
I had a thought - this won't work unless the type is declared in the function we're currently visiting, right? I.e. what if we have a type declaration like `type MySlice1 []string` in package scope rather than local scope? I think this solution will work in the latter case, but I feel like these typedefs are predominantly going to be in package scope

Reply:
was a bit busy with office work. I have added a bit of code to handle this now and tested.

Reply:
No worries! Looks like it's picking up slice typedefs now inside of the standard library :) 
`src/cmd/trace/goroutines.go:89 Consider preallocating glist`

Reply:
oh is it? how can i reproduce this?